### PR TITLE
docs(typescript): improve typescript docs and add peer deps

### LINKED ISSUE #10668: model type can not be infer because of the lack of @types/bluebird
## What are you doing?

I am using `sequelize` with typescript. Below is my demo file
```js
// code here
import {Sequelize, Model, STRING, DATE} from 'sequelize'
const sequelize = new Sequelize('postgres://user:pass@example.com:5432/dbname');

class User extends Model {}
User.init({
  username: STRING,
  birthday: DATE
}, { sequelize });

User.create({
  username: 'janedoe',
  birthday: new Date(1980, 6, 20)
}).then(vvvv => {

})
```

## What do you expect to happen?

`vvvv` should be infer to type `User`, but it doesn't.

## What is actually happening?

`vvvv` is infer to `any`.

when I run `npm i @types/bluebird`, the infer works.

__Dialect:__ any
__Dialect version:__ any
__Database version:__ any
__Sequelize version:__ 5.2.7
__Tested with latest release:__ Yes. 5.2.7

### PR DESCRIPTION
**Pull request by SimonSchick**
### Pull Request check-list

_Please make sure to review and check all of these items:_

- [ ] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [x] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [ ] Have you added new tests to prevent regressions?
- [x] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [x] Did you follow the commit message conventions explained in [CONTRIBUTING.md](https://github.com/sequelize/sequelize/blob/master/CONTRIBUTING.md)?

### Description of change

Fixes #10668

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -85,6 +85,11 @@
     "typescript": "^3.3.3333",
     "semantic-release": "^15.13.3"
   },
+  "peerDependencies": {
```
Comment (sushantdhiman) [inline package.json]: Hmm, what about non-typescript users. They shouldn't be forced to add these dependencies to their project.

Comment (SimonSchick) [reply package.json]: That was my thought too, that's why I didn't add them initially, but we've had a few users (see issue) complain about 'weird' errors due to these packages not being installed.

`peerDependencies` don't automatically install but give you a small warning which you can ignore, might be a fair trade off.

Comment (papb) [reply package.json]: @SimonSchick @sushantdhiman I am starting to think that `peerDependencies` is a bad idea too. I wouldn't like to see these as a non-TS user. What if we add a script to run once sequelize is loaded, to auto detect if typescript is being used, check if the dependencies are installed and print a warning message automatically on the console if they're missing?

Comment (papb) [reply package.json]: Also, about "some users complaining", I think that once the rest of this PR is merged (without peerDependencies), the documentation will be much clearer on TypeScript usage and these mistakes will be less frequent. How about just removing the `peerDependencies` thing from this PR and waiting a bit to see if the confusion remains?

Comment (papb) [reply package.json]: Someone else just had the same problem. I think this PR should be merged ASAP - but without the `peerDependencies` part. If the confusion persists, then we can think about it again.

(this is just my opinion, of course)

Comment (sushantdhiman) [reply package.json]: @SimonSchick remove peer dependencies and merge it

**Pull request merged**